MessagePackExtension default initialization unavailable

## Problem

It’s not possible for a client of the framework to initialize a `MessagePackExtension` with a `type` and `data`; it results in a compile error. There is a workaround, though.

## Example

The following code, compiled outside the MessagePacker package:

```swift
import MessagePacker

var ext = MessagePackExtension(type: Int8(1), data: Data())
```

will fail to compile with these errors:

```
/Users/snej/Projects/BLIP-Swift/BLIP-Swift/Capability.swift:50:19: error: no exact matches in call to initializer 
        var ext = MessagePackExtension(type: Int8(1), data: Data())
                  ^
/Users/snej/Library/Developer/Xcode/DerivedData/BLIP-Swift-ertqqmqyjxcexgamkgpqvrypkloh/SourcePackages/checkouts/MessagePacker/Sources/MessagePackExtension.swift:17:12: note: candidate expects value of type 'MessagePackTimestamp' for parameter #1 (got '(type: Int8, data: @lvalue Data)')
    public init(timestamp: MessagePackTimestamp) {
           ^
/Users/snej/Library/Developer/Xcode/DerivedData/BLIP-Swift-ertqqmqyjxcexgamkgpqvrypkloh/SourcePackages/checkouts/MessagePacker/Sources/MessagePackExtension.swift:36:12: note: candidate expects value of type 'any Decoder' for parameter #1 (got '(type: Int8, data: @lvalue Data)')
    public init(from decoder: Decoder) throws {
           ^
```

The compiler sees the two explicit initializers in extensions, and insists on using one of those.

The problem, I think, is that MessagePacker doesn’t have any explicit `(Int8, Data)` initializer function. Inside the framework you can use the implicit initializer (e.g. at the bottom of MessagePackType+ExtensionType.swift), but for some reason outside the framework you can’t.

## Fix

Explicitly implementing the `(Int8, Data)` initializer should fix this.

## Workaround

```swift
var ext = MessagePackExtension(timestamp: MessagePackTimestamp(date: Date.now))
ext.data = myData
ext.type = myType
```

--- a/Sources/MessagePackExtension.swift
+++ b/Sources/MessagePackExtension.swift
@@ -11,6 +11,11 @@ import Foundation
 public struct MessagePackExtension: Equatable {
     public var type: Int8
     public var data: Data
+    
+    public init(type: Int8, data: Data) {
+        self.type = type
+        self.data = data
+    }
 }
 
 extension MessagePackExtension {

--- a/Sources/MessagePackTimestamp.swift
+++ b/Sources/MessagePackTimestamp.swift
@@ -11,6 +11,11 @@ import Foundation
 public struct MessagePackTimestamp: Equatable {
     public var seconds: Int64
     public var nanoseconds: Int64
+    
+    public init(seconds: Int64, nanoseconds: Int64) {
+        self.seconds = seconds
+        self.nanoseconds = nanoseconds
+    }
 }
 
 extension MessagePackTimestamp {

